### PR TITLE
Create mesos documentation directory

### PR DESCRIPTION
**Pull request by L-MA**
Includes README file and Spike's Dockerized Guide moved from the calico-mesos repo.

### CODE REVIEW

File: docs/mesos/PrepareExternalServices.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Prepare External Services for Mesos + Calico Deployment
+
+## Docker
+
+- **Docker must be installed on each Mesos Agent** that is deploying calico via the packaged Calico container (recommended). Users who prefer to [run Calico as a baremetal service](#) do not need to install Docker on each Agent.
+- **Docker must be installed on Mesos Master** if etcd and ZooKeeper are being deployed as Docker containers (as this tutorial does). Users who are running etcd / ZooKeeper elsewhere do not need to install Docker on each Master.
```
Comment (djosborne) [inline docs/mesos/PrepareExternalServices.md]: remove `(as this tutorial does)`

**Comment by djosborne** [comment]
I think the etcd/zookeeper location disclaimer from your Centos7 rpm installation guide should get placed in either README.md, the PrepareExternalServices.md, or both? 

I'd argue that we should paste it into the top of PrepareExternalServices, as we want to keep the README as lean as possible.

https://github.com/projectcalico/calico-mesos/pull/19/files#diff-eaed77e9b60ec9f1c87033fb90e06fbbR5

File: docs/mesos/PrepareExternalServices.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    $ sudo docker pull jplock/zookeeper:3.4.5
+    $ sudo docker run --detach --name zookeeper -p 2181:2181 jplock/zookeeper:3.4.5
+
+## etcd
+Calico uses etcd as its data store and communication mechanism among Calico components.
+
+etcd needs your fully qualified domain name to start correctly.
+
+    $ sudo docker pull quay.io/coreos/etcd:v2.2.0
```
Comment (djosborne) [inline docs/mesos/PrepareExternalServices.md]: nix the `sudo` on these commands

File: docs/mesos/PrepareExternalServices.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    $ export FQDN=`hostname -f`
+    $ sudo mkdir -p /var/etcd
+    $ sudo docker run --detach --name etcd --net host -v /var/etcd:/data quay.io/coreos/etcd:v2.2.0 \
```
Comment (djosborne) [inline docs/mesos/PrepareExternalServices.md]: Change this to `$ sudo FQDN='hostname-f' docker run --detach ....` (but replace the single quote with a bash backtick)

Comment (L-MA) [inline docs/mesos/PrepareExternalServices.md]: nice catch

File: docs/mesos/PrepareHosts.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+You will either need to configure the firewalls on each node in your cluster (recommended) to allow access to the cluster services or disable it completely.  Included in this section is configuration examples for `firewalld`.  If you use a different firewall, check your documentation for how to open the listed ports.
+
+Master node(s) require
+
+| Service Name | Port/protocol     |
+|--------------|-------------------|
+| zookeeper    | 2181/tcp          |
+| mesos-master | 5050/tcp          |
+| etcd         | 2379/tcp 4001/tcp |
+| marathon     | 8080/tcp          |
```
Comment (djosborne) [inline docs/mesos/PrepareHosts.md]: I'm not sure if we should tell them to open up marathon and etcd ports here, or in the setting up external services section...

File: docs/mesos/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Mesos with Calico Networking
+Calico provides networking for your Mesos cluster. These guides will get you started.
+
+## Preparing Hosts
+[This guide](PrepareHosts.md) will walk you through preparing each host in your cluster to be compatible with Calico and Mesos.
```
Comment (djosborne) [inline docs/mesos/README.md]: I think SEO hates links that say "This Link" yadda yadda. I wrote these in when I was mocking out the link layout, but lets try to rephrase this sentence to have the guide name be the link. Something like:
```
The [Host Preperation Guide] will walk you through the necessary preparations on each host to get your Masters and Agents ready for Calico + Mesos.
```

Similar (but different wording) for the Prepare External Services guide as well.

File: docs/mesos/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Mesos with Calico Networking
+Calico provides networking for your Mesos cluster. These guides will get you started.
+
+## Preparing Hosts
```
Comment (djosborne) [inline docs/mesos/README.md]: Lets rename this "Prepare Masters and Agents"

File: docs/mesos/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+[This guide](PrepareHosts.md) will walk you through preparing each host in your cluster to be compatible with Calico and Mesos.
+
+## Preparing External Services
+Calico utilizes etcd as its datastore, while Mesos utilizes ZooKeeper. [This guide](PrepareExternalServices.md) will walk you through setting up both services using Docker.
+
+## Mesos + Netmodules
+There are several ways to install Mesos with Net-Modules.
```
Comment (djosborne) [inline docs/mesos/README.md]: Lets add a quick one liner as to why we're doing this:
```
Calico works in conjunction with [Netmodules][net-modules], a Mesos Networking Module. The following guides will install Mesos with the necessary Netmodules libraries:
```

Also use the link structure that spike did to get that link to work as is: https://github.com/projectcalico/calico-mesos/blame/master/docs/deploying.md#L208

File: docs/mesos/PrepareHosts.md (hunk 4 above)
Comment (L-MA) [inline docs/mesos/PrepareHosts.md]: Perhaps add a disclaimer in the PrepareExternalServices doc - "etcd uses tcp over ports 2379 and 4001. If you have a firewall configured on the host running etcd, open ports 2379 and 4001 to allow incoming and outgoing traffic."
Same for zk and marathon sections

File: docs/mesos/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+## Preparing External Services
+Calico utilizes etcd as its datastore, while Mesos utilizes ZooKeeper. [This guide](PrepareExternalServices.md) will walk you through setting up both services using Docker.
+
+## Mesos + Netmodules
+There are several ways to install Mesos with Net-Modules. 
+* Download and Install the Mesos + Net-Modules RPM (coming soon)
+* Create and Install the Mesos + Net-Modules RPM (coming soon)
+* Manually Compile and Install Mesos + Net-Modules (coming soon)
+
+## Calico
```
Comment (djosborne) [inline docs/mesos/README.md]: And a quick one liner here as to why we're doing this:
```
The following guides will help you install the Calico services required for each Slave in your Mesos Cluster:
```

File: docs/mesos/PrepareHosts.md (hunk 4 above)
Comment (djosborne) [inline docs/mesos/PrepareHosts.md]: That sounds good to me. I think we should remove the entries for etcd and marathon in this table and add a very similar disclaimer at the bottom of this section: `If you are running etcd, zookeeper, or Marathon on your Master, be sure to open ....`

Comment (djosborne) [inline docs/mesos/PrepareHosts.md]: Also add a link to the disclaimer on the external services section which takes you to this section.

**Comment by djosborne** [comment]
Can you make it more apparent that each of the 3 bullets in each section are optional paths? `Follow one of the following guides which best suits your requirements:`, or something along those lines, for each?

File: docs/mesos/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# Mesos with Calico Networking
+Calico provides networking for your Mesos cluster. These guides will get you started.
```
Comment (djosborne) [inline docs/mesos/README.md]: Could use a little friendlier welcome here, mirroring the calico-mesos repo:

**Calico provides IP-Per-Container Networking for your Mesos Cluster.** The following collection of guides will walk you through the steps necessary to get up and running.

File: docs/mesos/DockerizedDeployment.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+In these brief instructions, I'll concentrate on getting Mesos and Calico up and running as quickly as possible.  This means leaving out the details of how to configure higly-available services.  Instead, we'll install Zookeeper, etcd, and the Mesos Master on the same "master" node.
+
+
+# Preparation
+
+These instructions are designed to run on CentOS/Red Hat Enterprise Linux 7, but other than the initial commands to configure Docker, should work on any Linux distribution that supports Docker 1.7+ and `systemd`.
+
+If your distribution does not support `systemd`, you will need to create initialization files for each of the services.  These should be straightforward based on the included `.service` files, but talk to us on the [Calico Users' Slack](https://calicousers-slackin.herokuapp.com/) if you want some assistance.  If you write init files for a new system, share the love!  PRs are welcome :) 
+
+## Install Docker
```
Comment (djosborne) [inline docs/mesos/DockerizedDeployment.md]: I need to think about this more, so don't jump on this yet, but -

It doesn't make much sense that we're instructing users on how to install docker on their own systems. Realistically we should just link them to the docker install information for centos. This step made sense within Spike's guide as he was covering every single step between bare metal and a running cluster. But in these guides, I'm not sure it makes as much sense. 

The only reason I'm cautious to remove it is that the official docker docs don't mention anything about `docker-selinux`, as its a centos maintained rpm. And I can't find any documentation from centos covering how to install it...

File: docs/mesos/PrepareExternalServices.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
+# Prepare External Services for Mesos + Calico Deployment
+*Most Mesos clusters will run these services outside of their core Mesos cluster, separate from the Masters and Agents which read from them.*
+
+## Docker
+
+- **Docker must be installed on each Mesos Agent** that is deploying calico via the packaged Calico container (recommended). Users who prefer to [run Calico as a baremetal service](#) do not need to install Docker on each Agent.
+- **Docker must be installed on Mesos Master** if etcd and ZooKeeper are being deployed as Docker containers. Users who are running etcd / ZooKeeper elsewhere do not need to install Docker on each Master.
+
+Run the following commands to install Docker:
```
Comment (djosborne) [inline docs/mesos/PrepareExternalServices.md]: I need to think about this more, so don't jump on this yet, but -

It doesn't make much sense that we're instructing users on how to install docker on their own systems. Realistically we should just link them to the docker install information for centos. This step made sense within Spike's guide as he was covering every single step between bare metal and a running cluster. But in these guides, I'm not sure it makes as much sense. 

The only reason I'm cautious to remove it is that the official docker docs don't mention anything about `docker-selinux`, as its a centos maintained rpm. And I can't find any documentation from centos covering how to install it...

File: docs/mesos/README.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+* Download and Install the Mesos + Net-Modules RPM (coming soon)
+* Create and Install the Mesos + Net-Modules RPM (coming soon)
+* Manually Compile and Install Mesos + Net-Modules (coming soon)
+
+## Calico
+The following guides will help you install the Calico services required for each agent in your Mesos cluster.
+
+### Calico With Docker (Recommended)
+Calico is primarily distributed as a Docker container. The following guides will help you set that up.
+* Download and Install the Calico RPM (coming soon)
```
Comment (djosborne) [inline docs/mesos/README.md]: Lets mark this as `(Recommended)` as well

File: docs/mesos/PrepareHosts.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+| BIRD (BGP)   | 179/tcp           |
+| mesos-agent  | 5051/tcp          |
+
+Example `firewalld` config
+
+    $ sudo firewall-cmd --zone=public --add-port=179/tcp --permanent
+    $ sudo firewall-cmd --zone=public --add-port=5051/tcp --permanent
+    $ sudo systemctl restart firewalld
+
+*Note: etcd uses tcp over ports 2379 and 4001, ZooKeeper uses tcp over port 2181, and Marathon uses tcp over port 8080. If you choose to configure a firewall, be sure to open these ports on the host(s) running the service(s).*
```
Comment (djosborne) [inline docs/mesos/PrepareHosts.md]: change this to `If you choose to run these services on your Mesos-Master (as we do in the [External Services Guide]), be sure to open these ports on that host as well.`

File: docs/mesos/PrepareExternalServices.md (hunk 11 above)
Comment (djosborne) [inline docs/mesos/PrepareExternalServices.md]: I've thought about it a bit, and I think we should leave this in as is, and work towards removing it later on

File: docs/mesos/README.md (hunk 12 above)
Comment (L-MA) [inline docs/mesos/README.md]: Should we wait to mark it recommended until we have a guide and release with the calico-mesos rpms?

File: docs/mesos/README.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+The following guides will help you install the Calico services required for each agent in your Mesos cluster.
+
+### Calico With Docker (Recommended)
+Calico is primarily distributed as a Docker container. Follow one of the guides below to set that up.
+* Download and Install the Calico RPM (coming soon)
+* Create and Install the Calico RPM (coming soon)
+* Manually Install Calico (coming soon)
+
+### Calico Without Docker
+Don't want to use Docker in your Mesos Cluster? Calico can run on Baremetal. Choose one of the following guides to install Calico without Docker.
```
Comment (djosborne) [inline docs/mesos/README.md]: I probably wrote this in - we don't mean "Baremetal". Change it to `Calico can run directly on your Agent`.

File: docs/mesos/PrepareHosts.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+| BIRD (BGP)   | 179/tcp           |
+| mesos-agent  | 5051/tcp          |
+
+Example `firewalld` config
+
+    $ sudo firewall-cmd --zone=public --add-port=179/tcp --permanent
+    $ sudo firewall-cmd --zone=public --add-port=5051/tcp --permanent
+    $ sudo systemctl restart firewalld
+
+*Note: etcd uses tcp over ports 2379 and 4001, ZooKeeper uses tcp over port 2181, and Marathon uses tcp over port 8080. If you choose to run these services on your Mesos Master (as we do in the [External Services Guide](PrepareExternalServices.md), be sure to open these ports on that host as well.*
```
Comment (djosborne) [inline docs/mesos/PrepareHosts.md]: Missing an end parenthesis here.

File: docs/mesos/DockerizedDeployment.md [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+
+    $ sudo docker pull jplock/zookeeper:3.4.5
+    $ sudo docker pull quay.io/coreos/etcd:v2.2.0
+    $ sudo docker pull calico/mesos-calico
+    $ sudo docker pull mesosphere/marathon:v0.9.1
+    $ sudo docker pull calico/node:v0.8.0
+
+Next, download the unit files
+
+    $ wget https://github.com/projectcalico/calico-mesos/releases/0.1/units.tgz
```
Comment (spikecurtis) [inline docs/mesos/DockerizedDeployment.md]: this link needs to be updated

Comment (spikecurtis) [inline docs/mesos/DockerizedDeployment.md]: `tar xzf units.tgz`

File: docs/mesos/PrepareExternalServices.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
+# Prepare External Services for Mesos + Calico Deployment
+*Most Mesos clusters will run these services outside of their core Mesos cluster, separate from the Masters and Agents which read from them.*
```
Comment (spikecurtis) [inline docs/mesos/PrepareExternalServices.md]: "outside" the cluster sounds a bit strange.  I'd say "Most Mesos clusters will run these services on specific, dedicated machines chosen to maximize availability."

File: docs/mesos/README.md [hunk 18]
```diff
@@ -0,0 +1,34 @@
+# Mesos with Calico Networking
+**Calico provides IP-Per-Container networking for your Mesos cluster.** The following collection of guides will walk you through the steps necessary to get up and running.
+
+## Prepare Master and Agent Nodes
+[The Host Preparation Guide](PrepareHosts.md) will walk you through preparing each host in your cluster to be compatible with Calico and Mesos.
+
+## Prepare External Services
```
Comment (spikecurtis) [inline docs/mesos/README.md]: foundational services?
core services?

File: docs/mesos/README.md [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+* Download and Install the Mesos + Net-Modules RPM (coming soon)
+* Create and Install the Mesos + Net-Modules RPM (coming soon)
+* Manually Compile and Install Mesos + Net-Modules (coming soon)
+
+## Calico
+The following guides will help you install the Calico services required for each agent in your Mesos cluster.
+
+### Calico With Docker (Recommended)
+Calico is primarily distributed as a Docker container. Follow one of the guides below to set that up.
+* Download and Install the Calico RPM (coming soon)
```
Comment (djosborne) [inline docs/mesos/README.md]: These should be called `Calico-Mesos RPM`

File: docs/mesos/DockerizedDeployment.md [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+In these instructions, we will run all cluster services as [Docker][docker] containers.  This speeds deployment and will prevent pesky issues like incompatible dependencies.  The services we will need are
+
+ * Zookeeper
+ * etcd
+ * Mesos Master
+ * Mesos Agent
+ * Calico
+ * Marathon (Mesos framework)
+
+In these brief instructions, I'll concentrate on getting Mesos and Calico up and running as quickly as possible.  This means leaving out the details of how to configure higly-available services.  Instead, we'll install Zookeeper, etcd, and the Mesos Master on the same "master" node.
```
Comment (robbrockbank) [inline docs/mesos/DockerizedDeployment.md]: higly -> highly

Comment (robbrockbank) [inline docs/mesos/DockerizedDeployment.md]: I'll -> We'll

File: docs/mesos/DockerizedDeployment.md [hunk 21]
```diff
@@ -0,0 +1,210 @@
...
+
+
+# Preparation
+
+These instructions are designed to run on CentOS/Red Hat Enterprise Linux 7, but other than the initial commands to configure Docker, should work on any Linux distribution that supports Docker 1.7+ and `systemd`.
```
Comment (robbrockbank) [inline docs/mesos/DockerizedDeployment.md]: Linux 7, but  ->  Linux 7 but,

File: docs/mesos/README.md (hunk 8 above)
Comment (robbrockbank) [inline docs/mesos/README.md]: We should also have a link to this document from the main README.md

**Comment by robbrockbank** [comment]
LGTM from a general doc perspective.  Has it been reviewed from a Mesos perspective?

**Comment by spikecurtis** [comment]
LGTM

**Pull request merged**